src/pantry.js

```diff
@@ -1,7 +1,75 @@
 class Pantry {
-  constructor(userIngredients) {
-    this.contents = userIngredients;
+  constructor(userIngredients, ingredientsData) {
+    this.contents = userIngredients,
+    this.ingredientsData = ingredientsData
   }
+
+  compareRecipeToPantryIngredients(chosenRecipe) {
+
+    let ingredientDetails = chosenRecipe.ingredients.map(ingredient => {
+      return {
+        name: ingredient.name,
+        id: ingredient.id,
+        qty: ingredient.quantity.amount,
+        unit: ingredient.quantity.unit,
+        cost: this.ingredientsData.find(ingredientInfo => {
+          return ingredientInfo.id === ingredient.id
+        }).estimatedCostInCents
+      };
+    });
+
+    ingredientDetails.forEach(ingredient => {
+      let pantryMatch = this.contents.find(item => {
+        if (ingredient.id === item.ingredient) {
+          ingredient.qty -= item.amount
+        };
+      });
+    });
+
+    let ingredientsNeeded = [];
+    ingredientDetails.forEach(ingredient => {
+      if (ingredient.qty > 0) {
+        ingredientsNeeded.push(ingredient)
+      };
+    });
+    return ingredientsNeeded;
```

Comment:
Overall, all the methods looks good.  I think that in this section to refactor, we can use a Reduce iterator instead of a forEach iterator.  With the Reduce iterator, we can assign the return value into a [ ] in the reduce.  For example: let ingredientsNeeded = ingredientDetails.reduce(ingredient => { .... },[]).

Reply:
That's a good idea, I'll refactor it using reduce later today